src/modules/song/thunks.ts

```diff
@@ -277,25 +291,67 @@ export const patchSong = createAsyncThunk(
         for (const collabResp of updateCollabResponses) {
           if ("error" in collabResp) return;
         }
-      }
 
-      if (body.title && body.artistName) {
+        const songResp = await dispatch(
+          songApi.endpoints.getSong.initiate(body.id)
+        );
+
+        if ("error" in songResp || !songResp.data) return;
+
+        const profileResp = await dispatch(
+          sessionApi.endpoints.getProfile.initiate()
+        );
+
+        if ("error" in profileResp || !profileResp.data) return;
+
+        // if body.isMinting is present and true, then the song is being
+        // minted for the first time, save the artist agreement.
```

Comment:
was this comment left by accident or out of place? i dont see "body.isMinting" check nearby